### PR TITLE
home-assistant: 2021.5.5 -> 2021.6.3

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

https://www.home-assistant.io/blog/2021/06/02/release-20216/

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125445` at dcdb6d2d run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>22 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/915c694b65c48bda0b68a5b45448c8f1">chia</a></li>
    <li><a href="https://gist.github.com/25f8c8deaf4bf82ea29315791c13f6d0">dyndnsc</a></li>
    <li><a href="https://gist.github.com/88554e7c644e0ffc1bc82aca5624c397">python38Packages.entrance</a></li>
    <li><a href="https://gist.github.com/de8a5d0c943b2d5d4bf123b499cefd7f">python38Packages.entrance-with-router-features</a></li>
    <li><a href="https://gist.github.com/7c5afb4294c190eb4702aa64da745d8a">python38Packages.hbmqtt</a></li>
    <li><a href="https://gist.github.com/0f7a70680e3f7c77ec1c19ad387a28a9">python38Packages.json-logging</a></li>
    <li><a href="https://gist.github.com/f872dd70f0ddb41efc0c388de8d9ee83">python38Packages.pytest-sanic</a></li>
    <li><a href="https://gist.github.com/e5127fbd76af851e6b2726eb397ae2e1">python38Packages.python-smarttub</a></li>
    <li><a href="https://gist.github.com/c41e9edd8d65aa9c3957e0368e2f4b2d">python38Packages.sanic</a></li>
    <li><a href="https://gist.github.com/09c945a97e94abbbee67c38bb7e6608a">python38Packages.sanic-auth</a></li>
    <li><a href="https://gist.github.com/420b9c62882664067dfdc1c6a061cd46">python38Packages.sanic-testing</a></li>
    <li><a href="https://gist.github.com/ddf3db56c8eabd5fe2f262d67f4ba1ea">python39Packages.authcaptureproxy</a></li>
    <li><a href="https://gist.github.com/a00989a921a884c8a5626071f4ff2d94">python39Packages.entrance</a></li>
    <li><a href="https://gist.github.com/9313132eafd8ac058b9d895b4712a728">python39Packages.entrance-with-router-features</a></li>
    <li><a href="https://gist.github.com/e2a4ddaf1a3ccfeed2fa8a65cbec8d8b">python39Packages.json-logging</a></li>
    <li><a href="https://gist.github.com/df2e4b6c63c5eb6c44acc629fa4800b9">python39Packages.pytest-sanic</a></li>
    <li><a href="https://gist.github.com/a98b26b26a90be360d64f7b666177b71">python39Packages.python-smarttub</a></li>
    <li><a href="https://gist.github.com/25f20558a3c2eaa8f49d9ac95b2d595b">python39Packages.qiskit</a></li>
    <li><a href="https://gist.github.com/72b15f3815539b859d50e95d5ff1c384">python39Packages.qiskit-ibmq-provider</a></li>
    <li><a href="https://gist.github.com/2063fcf0d6170ae75b288baf172c87f1">python39Packages.sanic</a></li>
    <li><a href="https://gist.github.com/d37cb5da4f1b69ae5b2eb901d2ac0172">python39Packages.sanic-auth</a></li>
    <li><a href="https://gist.github.com/1e2bd1127ed0f3cde9436b4683bb0676">python39Packages.sanic-testing</a></li>
  </ul>
</details>
<details>
  <summary>7 packages skipped due to time constraints:</summary>
  <ul>
    <li>home-assistant</li>
    <li>python38Packages.qiskit</li>
    <li>python38Packages.qiskit-ibmq-provider</li>
    <li>python38Packages.slack-sdk</li>
    <li>python39Packages.slack-sdk</li>
    <li>python39Packages.teslajsonpy</li>
    <li>theharvester</li>
  </ul>
</details>
<details>
  <summary>107 packages built successfully:</summary>
  <ul>
    <li>appdaemon</li>
    <li>datasette (python38Packages.datasette)</li>
    <li>esphome</li>
    <li>grab-site</li>
    <li>onionshare</li>
    <li>onionshare-gui</li>
    <li>opsdroid</li>
    <li>platformio</li>
    <li>python38Packages.PyRMVtransport</li>
    <li>python38Packages.aioambient</li>
    <li>python38Packages.aiohomekit</li>
    <li>python38Packages.aiohue</li>
    <li>python38Packages.aiopvpc</li>
    <li>python38Packages.aiopylgtv</li>
    <li>python38Packages.amqtt</li>
    <li>python38Packages.authcaptureproxy</li>
    <li>python38Packages.axis</li>
    <li>python38Packages.boschshcpy</li>
    <li>python38Packages.bravia-tv</li>
    <li>python38Packages.denonavr</li>
    <li>python38Packages.dependency-injector</li>
    <li>python38Packages.elmax</li>
    <li>python38Packages.envoy-reader</li>
    <li>python38Packages.fastapi</li>
    <li>python38Packages.flask-socketio</li>
    <li>python38Packages.garages-amsterdam</li>
    <li>python38Packages.gios</li>
    <li>python38Packages.growattserver</li>
    <li>python38Packages.ha-philipsjs</li>
    <li>python38Packages.hap-python</li>
    <li>python38Packages.homematicip</li>
    <li>python38Packages.httpcore</li>
    <li>python38Packages.httpx</li>
    <li>python38Packages.ipfshttpclient</li>
    <li>python38Packages.ismartgate</li>
    <li>python38Packages.mattermostdriver</li>
    <li>python38Packages.nexia</li>
    <li>python38Packages.openhomedevice</li>
    <li>python38Packages.pyatmo</li>
    <li>python38Packages.pydroid-ipcam</li>
    <li>python38Packages.pylitterbot</li>
    <li>python38Packages.pyppeteer</li>
    <li>python38Packages.pytest-httpx</li>
    <li>python38Packages.python-binance</li>
    <li>python38Packages.python-engineio</li>
    <li>python38Packages.python-engineio_3</li>
    <li>python38Packages.python-socketio</li>
    <li>python38Packages.python-socketio_4</li>
    <li>python38Packages.respx</li>
    <li>python38Packages.roombapy</li>
    <li>python38Packages.roonapi</li>
    <li>python38Packages.simplisafe-python</li>
    <li>python38Packages.systembridge</li>
    <li>python38Packages.teslajsonpy</li>
    <li>python38Packages.uvicorn</li>
    <li>python38Packages.venstarcolortouch</li>
    <li>python38Packages.websockets</li>
    <li>python38Packages.zwave-js-server-python</li>
    <li>python39Packages.PyRMVtransport</li>
    <li>python39Packages.aioambient</li>
    <li>python39Packages.aiohue</li>
    <li>python39Packages.aiopvpc</li>
    <li>python39Packages.aiopylgtv</li>
    <li>python39Packages.amqtt</li>
    <li>python39Packages.axis</li>
    <li>python39Packages.boschshcpy</li>
    <li>python39Packages.bravia-tv</li>
    <li>python39Packages.datasette</li>
    <li>python39Packages.denonavr</li>
    <li>python39Packages.dependency-injector</li>
    <li>python39Packages.elmax</li>
    <li>python39Packages.envoy-reader</li>
    <li>python39Packages.fastapi</li>
    <li>python39Packages.flask-socketio</li>
    <li>python39Packages.garages-amsterdam</li>
    <li>python39Packages.gios</li>
    <li>python39Packages.growattserver</li>
    <li>python39Packages.ha-philipsjs</li>
    <li>python39Packages.hap-python</li>
    <li>python39Packages.homematicip</li>
    <li>python39Packages.httpcore</li>
    <li>python39Packages.httpx</li>
    <li>python39Packages.ipfshttpclient</li>
    <li>python39Packages.ismartgate</li>
    <li>python39Packages.mattermostdriver</li>
    <li>python39Packages.nexia</li>
    <li>python39Packages.openhomedevice</li>
    <li>python39Packages.pyatmo</li>
    <li>python39Packages.pydroid-ipcam</li>
    <li>python39Packages.pylitterbot</li>
    <li>python39Packages.pyppeteer</li>
    <li>python39Packages.pytest-httpx</li>
    <li>python39Packages.python-binance</li>
    <li>python39Packages.python-engineio</li>
    <li>python39Packages.python-engineio_3</li>
    <li>python39Packages.python-socketio</li>
    <li>python39Packages.python-socketio_4</li>
    <li>python39Packages.respx</li>
    <li>python39Packages.roombapy</li>
    <li>python39Packages.roonapi</li>
    <li>python39Packages.simplisafe-python</li>
    <li>python39Packages.systembridge</li>
    <li>python39Packages.uvicorn</li>
    <li>python39Packages.venstarcolortouch</li>
    <li>python39Packages.websockets</li>
    <li>python39Packages.zwave-js-server-python</li>
    <li>urlwatch</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/development/python-modules/pydroid-ipcam/default.nix:5:3:
```
  |
5 | , fetchpatch
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/development/python-modules/aiopvpc/default.nix:7:3:
```
  |
7 | , fetchpatch
  |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/websockets/default.nix:32:3:
```
   |
32 |   meta = with lib; {
   |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by jonringer** [comment]
@GrahamcOfBorg test homeassistant

**Comment by jonringer** [comment]
@GrahamcOfBorg test home-assistant

**Comment by fabaff** [comment]
https://github.com/NixOS/nixpkgs/pull/123283 is already covering the `websockets` and the `systembridge` update.

**Comment by mweinelt** [comment]
I'll rebase as soon as yours is merged then.

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 125445` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.python-smarttub</li>
    <li>python39Packages.python-smarttub</li>
  </ul>
</details>
<details>
  <summary>19 packages failed to build:</summary>
  <ul>
    <li>chia</li>
    <li>dyndnsc</li>
    <li>grab-site</li>
    <li>python38Packages.entrance</li>
    <li>python38Packages.entrance-with-router-features</li>
    <li>python38Packages.hbmqtt</li>
    <li>python38Packages.json-logging</li>
    <li>python38Packages.pytest-sanic</li>
    <li>python38Packages.sanic</li>
    <li>python38Packages.sanic-auth</li>
    <li>python38Packages.sanic-testing</li>
    <li>python39Packages.entrance</li>
    <li>python39Packages.entrance-with-router-features</li>
    <li>python39Packages.json-logging</li>
    <li>python39Packages.pytest-sanic</li>
    <li>python39Packages.sanic</li>
    <li>python39Packages.sanic-auth</li>
    <li>python39Packages.sanic-testing</li>
    <li>theharvester</li>
  </ul>
</details>
<details>
  <summary>113 packages built:</summary>
  <ul>
    <li>appdaemon</li>
    <li>datasette (python38Packages.datasette)</li>
    <li>esphome</li>
    <li>home-assistant</li>
    <li>onionshare</li>
    <li>onionshare-gui</li>
    <li>opsdroid</li>
    <li>platformio</li>
    <li>python38Packages.HAP-python (python38Packages.hap-python)</li>
    <li>python38Packages.PyRMVtransport</li>
    <li>python38Packages.aioambient</li>
    <li>python38Packages.aiohomekit</li>
    <li>python38Packages.aiohue</li>
    <li>python38Packages.aiopylgtv</li>
    <li>python38Packages.amqtt</li>
    <li>python38Packages.authcaptureproxy</li>
    <li>python38Packages.axis</li>
    <li>python38Packages.boschshcpy</li>
    <li>python38Packages.bravia-tv</li>
    <li>python38Packages.denonavr</li>
    <li>python38Packages.dependency-injector</li>
    <li>python38Packages.elmax</li>
    <li>python38Packages.envoy-reader</li>
    <li>python38Packages.fastapi</li>
    <li>python38Packages.flask-socketio</li>
    <li>python38Packages.garages-amsterdam</li>
    <li>python38Packages.gios</li>
    <li>python38Packages.growattserver</li>
    <li>python38Packages.ha-philipsjs</li>
    <li>python38Packages.homematicip</li>
    <li>python38Packages.httpcore</li>
    <li>python38Packages.httpx</li>
    <li>python38Packages.ipfshttpclient</li>
    <li>python38Packages.ismartgate</li>
    <li>python38Packages.mattermostdriver</li>
    <li>python38Packages.nexia</li>
    <li>python38Packages.openhomedevice</li>
    <li>python38Packages.pyatmo</li>
    <li>python38Packages.pydroid-ipcam</li>
    <li>python38Packages.pylitterbot</li>
    <li>python38Packages.pyppeteer</li>
    <li>python38Packages.pytest-httpx</li>
    <li>python38Packages.python-binance</li>
    <li>python38Packages.python-engineio</li>
    <li>python38Packages.python-engineio_3</li>
    <li>python38Packages.python-socketio</li>
    <li>python38Packages.python-socketio_4</li>
    <li>python38Packages.qiskit</li>
    <li>python38Packages.qiskit-ibmq-provider</li>
    <li>python38Packages.respx</li>
    <li>python38Packages.roombapy</li>
    <li>python38Packages.roonapi</li>
    <li>python38Packages.simplisafe-python</li>
    <li>python38Packages.slack-sdk</li>
    <li>python38Packages.systembridge</li>
    <li>python38Packages.teslajsonpy</li>
    <li>python38Packages.uvicorn</li>
    <li>python38Packages.venstarcolortouch</li>
    <li>python38Packages.websockets</li>
    <li>python38Packages.zwave-js-server-python</li>
    <li>python39Packages.HAP-python (python39Packages.hap-python)</li>
    <li>python39Packages.PyRMVtransport</li>
    <li>python39Packages.aioambient</li>
    <li>python39Packages.aiohue</li>
    <li>python39Packages.aiopylgtv</li>
    <li>python39Packages.amqtt</li>
    <li>python39Packages.authcaptureproxy</li>
    <li>python39Packages.axis</li>
    <li>python39Packages.boschshcpy</li>
    <li>python39Packages.bravia-tv</li>
    <li>python39Packages.datasette</li>
    <li>python39Packages.denonavr</li>
    <li>python39Packages.dependency-injector</li>
    <li>python39Packages.elmax</li>
    <li>python39Packages.envoy-reader</li>
    <li>python39Packages.fastapi</li>
    <li>python39Packages.flask-socketio</li>
    <li>python39Packages.garages-amsterdam</li>
    <li>python39Packages.gios</li>
    <li>python39Packages.growattserver</li>
    <li>python39Packages.ha-philipsjs</li>
    <li>python39Packages.homematicip</li>
    <li>python39Packages.httpcore</li>
    <li>python39Packages.httpx</li>
    <li>python39Packages.ipfshttpclient</li>
    <li>python39Packages.ismartgate</li>
    <li>python39Packages.mattermostdriver</li>
    <li>python39Packages.nexia</li>
    <li>python39Packages.openhomedevice</li>
    <li>python39Packages.pyatmo</li>
    <li>python39Packages.pydroid-ipcam</li>
    <li>python39Packages.pylitterbot</li>
    <li>python39Packages.pyppeteer</li>
    <li>python39Packages.pytest-httpx</li>
    <li>python39Packages.python-binance</li>
    <li>python39Packages.python-engineio</li>
    <li>python39Packages.python-engineio_3</li>
    <li>python39Packages.python-socketio</li>
    <li>python39Packages.python-socketio_4</li>
    <li>python39Packages.qiskit</li>
    <li>python39Packages.qiskit-ibmq-provider</li>
    <li>python39Packages.respx</li>
    <li>python39Packages.roombapy</li>
    <li>python39Packages.roonapi</li>
    <li>python39Packages.simplisafe-python</li>
    <li>python39Packages.slack-sdk</li>
    <li>python39Packages.systembridge</li>
    <li>python39Packages.teslajsonpy</li>
    <li>python39Packages.uvicorn</li>
    <li>python39Packages.venstarcolortouch</li>
    <li>python39Packages.websockets</li>
    <li>python39Packages.zwave-js-server-python</li>
    <li>urlwatch</li>
  </ul>
</details>

**Comment by mweinelt** [comment]
And fixed up aiomultiprocess, so now we're down to everything sanic.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -6226,8 +6234,6 @@ in {
 
   pyro-ppl = callPackage ../development/python-modules/pyro-ppl { };
 
-  pyroon = callPackage ../development/python-modules/pyroon { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:6229]: This should get an alias, too.

File: pkgs/servers/home-assistant/default.nix [hunk 2]
```diff
@@ -725,6 +761,9 @@ in with py.pkgs; buildPythonApplication rec {
     # put ping binary into PATH, e.g. for wake_on_lan tests
     export PATH=${inetutils}/bin:$PATH
 
+    # set up zoneinfo data for backports-zoneinfo in pvpc_hourly_pricing tests
+    export PYTHONTZPATH="${tzdata}/share/zoneinfo"
```
Comment (SuperSandro2000) [inline pkgs/servers/home-assistant/default.nix:765]: That only works for python 3.10.

File: pkgs/servers/home-assistant/default.nix [hunk 3]
```diff
@@ -215,7 +242,9 @@ in with py.pkgs; buildPythonApplication rec {
     voluptuous
     voluptuous-serialize
     yarl
-  ] ++ componentBuildInputs ++ extraBuildInputs;
+  ] ++ lib.optionals (pythonOlder "3.9") [
+    backports-zoneinfo
+  ]++ componentBuildInputs ++ extraBuildInputs;
```
Comment (SuperSandro2000) [inline pkgs/servers/home-assistant/default.nix:247]:
```suggestion
  ] ++ componentBuildInputs ++ extraBuildInputs;
```

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/top-level/python-packages.nix:6229]: I disagree. It was introduced on 2021-05-28 into master and home-assistant is its only consumer within nixpkgs. I think it is fair to say we can deprecate the wrong name with impunity.

File: pkgs/servers/home-assistant/default.nix (hunk 2 above)
Comment (mweinelt) [reply pkgs/servers/home-assistant/default.nix:765]: I somewhat expect #125599 to fix that.

**Comment by mweinelt** [comment]
https://github.com/home-assistant/core/releases/tag/2021.6.2

**Comment by mweinelt** [comment]
https://github.com/home-assistant/core/releases/tag/2021.6.3

**Comment by mweinelt** [comment]
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.python-smarttub</li>
    <li>python39Packages.python-smarttub</li>
  </ul>
</details>
<details>
  <summary>42 packages built:</summary>
  <ul>
    <li>home-assistant</li>
    <li>python38Packages.HAP-python (python38Packages.hap-python)</li>
    <li>python38Packages.aiohomekit</li>
    <li>python38Packages.aiohue</li>
    <li>python38Packages.aiolyric</li>
    <li>python38Packages.aiorecollect</li>
    <li>python38Packages.boschshcpy</li>
    <li>python38Packages.bravia-tv</li>
    <li>python38Packages.garages-amsterdam</li>
    <li>python38Packages.gios</li>
    <li>python38Packages.growattserver</li>
    <li>python38Packages.ismartgate</li>
    <li>python38Packages.nexia</li>
    <li>python38Packages.openhomedevice</li>
    <li>python38Packages.pyatmo</li>
    <li>python38Packages.pydroid-ipcam</li>
    <li>python38Packages.roonapi</li>
    <li>python38Packages.simplisafe-python</li>
    <li>python38Packages.systembridge</li>
    <li>python38Packages.venstarcolortouch</li>
    <li>python38Packages.xknx</li>
    <li>python38Packages.zwave-js-server-python</li>
    <li>python39Packages.HAP-python (python39Packages.hap-python)</li>
    <li>python39Packages.aiohue</li>
    <li>python39Packages.aiolyric</li>
    <li>python39Packages.aiorecollect</li>
    <li>python39Packages.boschshcpy</li>
    <li>python39Packages.bravia-tv</li>
    <li>python39Packages.garages-amsterdam</li>
    <li>python39Packages.gios</li>
    <li>python39Packages.growattserver</li>
    <li>python39Packages.ismartgate</li>
    <li>python39Packages.nexia</li>
    <li>python39Packages.openhomedevice</li>
    <li>python39Packages.pyatmo</li>
    <li>python39Packages.pydroid-ipcam</li>
    <li>python39Packages.roonapi</li>
    <li>python39Packages.simplisafe-python</li>
    <li>python39Packages.systembridge</li>
    <li>python39Packages.venstarcolortouch</li>
    <li>python39Packages.xknx</li>
    <li>python39Packages.zwave-js-server-python</li>
  </ul>
</details>

And smarttub isn't even broken in the home-assistant environment.

**Comment by fabaff** [comment]
@mweinelt, thanks for fixing `systembridge`. Was a remaining failure from #123283.

**Pull request merged**